csharp/src/Ice/Reference.cs

```diff
@@ -1315,10 +1376,9 @@ internal void Write(OutputStream ostr)
             _fixedConnection.ThrowException(); // Throw in case our connection is already destroyed.
             _requestHandler = _fixedConnection;
 
-            if (Protocol == Protocol.Ice2 && (byte)InvocationMode > (byte)InvocationMode.Oneway)
+            if (Protocol != Protocol.Ice1)
             {
-                throw new ArgumentException(
-                    $"invocation mode `{InvocationMode}' is not compatible with the ice2 protocol");
+                Debug.Assert((byte)InvocationMode <= (byte)InvocationMode.Oneway);
```

Comment:
I would remove the if and just check the protocol in the assert

```
Debug.Assert(Protocol == Protocol.Ice1 || (byte)InvocationMode <= (byte)InvocationMode.Oneway);
```

Reply:
Fixed as suggested.